packages/replay/src/session/Session.ts

```diff
@@ -24,59 +23,48 @@ interface SessionObject {
   segmentId: number;
 
   /**
-   * Is the session sampled? `null` if the sampled, otherwise, `session` or `error`
+   * The ID of the previous session.
+   * If this is empty, there was no previous session.
    */
-  sampled: Sampled;
-}
-
-export class Session {
-  /**
-   * Session ID
-   */
-  public readonly id: string;
+  previousSessionId?: string;
 
   /**
-   * Start time of current session
+   * Is the session sampled? `false` if not sampled, otherwise, `session` or `error`
    */
-  public started: number;
-
-  /**
-   * Last known activity of the session
-   */
-  public lastActivity: number;
-
-  /**
-   * Sequence ID specific to replay updates
-   */
-  public segmentId: number;
+  sampled: Sampled;
+}
 
-  /**
-   * Previous session ID
-   */
-  public previousSessionId: string | undefined;
+/**
+ * Get a session with defaults & applied sampling.
+ */
+export function makeSession(session: Partial<Session> & { sampled: Sampled }): Session {
+  const now = new Date().getTime();
+  const id = session.id || uuid4();
+  const started = session.started || now;
```

Comment:
Since tests are passing and I think for this case there is no difference, I guess the risk is low but just to confirm:

Is this change from the nullish coalescing operator safe here? `||` has different behaviour for falsy values than `??`. (but as I said, probably it's fine)

Reply:
I guess this behavior is technically not correct if `started` is 0, but I think that's an edge case we don't worry about.

Reply:
Exactly my thoughts

Reply:
Yes, technically it is not the same, but I think we don't care that you cannot set a time of January 1st 1979. There is actually a test that I adjusted for this. I will add a comment to this file as well explaining that this is a conscious decision!